### PR TITLE
chore(core): generate type file for icons

### PR DESCRIPTION
**Pull request by felipefialho**
## Infos

[Task](https://juntossomosmais.monday.com/boards/3957397225/pulses/5343021882)

#### What is being delivered?

- Generate `icons.d.ts` inside `core` to create a new release when add a new icon

#### What impacts?

We were having a problem involving the creation of a new release when we added a new icon, because the `core` folder was not modified.

### CODE REVIEW

File: packages/icons/package.json [hunk 1]
```diff
@@ -20,11 +20,6 @@
         "dependsOn": [
           "compile"
         ]
```
Comment (RodrigoRVSN) [inline packages/icons/package.json:22]: question: doesn't need to be changed [here](https://github.com/juntossomosmais/atomium/blob/main/packages/core/package.json#L40-L62) and [here](https://github.com/juntossomosmais/atomium/blob/main/apps/docs/package.json#L8-L31) too?

Comment (mauriciomutte) [reply packages/icons/package.json:22]: Yes, it will also be necessary too

Comment (felipefialho) [reply packages/icons/package.json:22]: thanks a lot guys! done at [d0cbd19](https://github.com/juntossomosmais/atomium/pull/289/commits/d0cbd19fbda0e24ca58f7732ec0c2cb8d2371870) and [7f1214f](https://github.com/juntossomosmais/atomium/pull/289/commits/7f1214fdd08a98797946e3b2fcc99e7eea04f26e)

**Review by RodrigoRVSN — APPROVED** [review]
lgtm

**Review by mauriciomutte — APPROVED** [review]

**Review by RodrigoRVSN — APPROVED** [review]

**Review by luiz-ortega — APPROVED** [review]

**Review by RodrigoRVSN — APPROVED** [review]

**Pull request merged**